frontend/src/AppHelpers.res

```diff
@@ -31,34 +31,48 @@ let sortedMetadata = dataByTestName => {
   ->Belt.SortArray.stableSortBy((a, b) =>
     compare(Js.Date.getTime(a["runAt"]), Js.Date.getTime(b["runAt"]))
   )
+  ->Belt.Array.map(md => (md["commit"], md["runAt"], md["run_job_id"]))
 }
 
 // Fill in NaN values and add missing commit metadata
-let addMissingCommits = ((metricTimeseries, metricMetadata), allCommits) => {
+let addMissingCommits = ((metricTimeseries, metricMetadata) as data, allCommits) => {
   let n = Belt.Array.length(allCommits)
   switch Belt.Array.length(metricTimeseries) < n {
-  | false => metricTimeseries
+  | false => data
+
   | true => {
+      // We use a NaN value to fill in values for missing commits
       let missingValue = [Js.Float._NaN, Js.Float._NaN, Js.Float._NaN]
-      let filledTimeseries = allCommits->Belt.Array.map(commit =>
+      let filledTimeseries = allCommits->Belt.Array.map(((commit, _, _)) =>
         switch metricMetadata->Belt.Array.getIndexBy(md => md["commit"] == commit) {
         | Some(idx) => Belt.Array.getExn(metricTimeseries, idx)
         | None => missingValue
         }
       )
-      filledTimeseries
+      // We use the last metadata item and use that as a template to create
+      // entries for missing commits. commit, runAt, lines and run_job_id are
+      // overwritten.  Other metadata like units, trend, description are the
+      // same as metadata in the last commit.
+      let templateMetadata = BeltHelpers.Array.lastExn(metricMetadata)
+      let filledMetadata =
+        allCommits->Belt.Array.map(((commit, runAt, run_job_id)) =>
+          Js.Obj.assign(
+            templateMetadata,
+            {"commit": commit, "runAt": runAt, "run_job_id": run_job_id, "lines": []},
```

Comment:
I'm not sure why we have a javascript object to represent all these stuff (I think `BenchmarkData.res` could have defined a standard record)... But now it's a bit everywhere and it's not really convenient.

For example, I don't see a nice alternative here to `Js.Obj.assign` and the issue is that this function breaks the typechecker (it returns an unknown object that could contain anything -- ie it's as bad as `Obj.magic`)

Reply:
I agree that it is annoying that the metric defined in `BenchmarkData.res` is not a standard record. I've felt that having a record would've been a lot more convenient atleast a couple of times. We should refactor that.